### PR TITLE
add scripts to allow for easier testing of aws postgres storage alerts

### PR DESCRIPTION
**Pull request by ciaranRoche**
Co-Authored-By: laurafitzgerald <lfitzger@redhat.com>

## Overview

Related to Jira: https://issues.redhat.com/browse/INTLY-7231

## Verification Postgres

- Clone this branch
- Run `make cluster/prepare`
- Run `make cluster/seed/managed/postgres`
- Run `make run`
- Follow `postgres/README.md` in PR to run `load.sh`
- Log into AWS, navigate to RDS console, and select created instance
- Click on Monitoring and watch the `FreeStorageSpace` graph.
- Ensure the graph trends down

## Verification Redis
- Clone this branch
- Run `make cluster/prepare`
- Run `make cluster/seed/managed/elasticache`
- Run `make run`
- Follow `redis/README.md` in PR to deploy `redis-load`
- Log into AWS, navigate to Elasticache console, and select created instance
- Click on primary node, and navigate to `Database Memory Usage Percentage` graph
- Ensure the graph trends up

### CODE REVIEW

File: hack/postgres/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+Scripts to load postgres (AWS) with data to easily allow and test alerting
+
+## Usage 
+Copy file to pod 
+```
+ oc cp load.sh <<namespace>>/<<pod name>>:/var/lib/pgsql
+```
+Run command
+``` 
+oc exec load-postgres-b54fffcf5-52cvf sh /var/lib/pgsql/load.sh <<host>> <<postgres password>> <<size to fill in GiB>>
```
Comment (psturc) [inline hack/postgres/README.md:12]:
```suggestion
oc exec <<pod name>> sh /var/lib/pgsql/load.sh <<host>> <<postgres password>> <<size to fill in GiB>> -n <<namespace>>
```

File: hack/postgres/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+```
```
Comment (psturc) [inline hack/postgres/README.md:13]: is it worth to add the "usage" guide for `clean.sh` script as well?

**Comment by psturc** [comment]
lgtm, just few comments, I'm going to verify this now

**Comment by aidenkeating** [comment]
LGTM, going to verify the Redis side now

File: hack/redis/redis-load-test.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                - name: PORT
+                  value: ${PORT}
+                - name: HOST
+                  value: ${HOST}
+                - name: ENTRIES
+                  value: ${ENTRIES}
+                - name: USERS
+                  value: ${USERS}
+              name: redis-load
+              image: quay.io/croche/redis-load:latest
```
Comment (aidenkeating) [inline hack/redis/redis-load-test.yaml:37]: Where is the source for it?

Comment (ciaranRoche) [reply hack/redis/redis-load-test.yaml:37]: Happy to move this, or make it better and move it into CRO if you want -> https://github.com/ciaranRoche/redis-load

Comment (aidenkeating) [reply hack/redis/redis-load-test.yaml:37]: I think it'd be good to pull this in. Mainly for future maintenance and so it doesn't get lost over time.

File: hack/redis/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+## Load Testing Redis AWS (Elasticache)
+The following is taking from (redis-load)[https://github.com/ciaranRoche/redis-load] which generates random entries to redis
+### Usage 
+#### Prerequisites
+```
+export ns=<<namespace you want to deploy the load generator>>
+```
+#### Run Load Generator
+``` 
+oc create -f redis-load.yaml -n $ns
```
Comment (aidenkeating) [inline hack/redis/README.md:14]:
```suggestion
oc create -f redis-load-test.yaml -n $ns
```

File: hack/redis/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+oc process redis-load-template -p HOST=<<redis host>> -p PORT=6379 -p ENTRIES=<<no of entries to create>> -p USERS=<<no of concurrent user>> | oc create -f -
```
Comment (aidenkeating) [inline hack/redis/README.md:15]: What values are recommended to be used here?

File: hack/redis/README.md (hunk 4 above)
Comment (aidenkeating) [reply hack/redis/README.md:14]: Actually, maybe we could avoid creating the template in OpenShift and do

```
export cr=<<name of redis cr>>

oc process -f redis-load-test.yaml -p HOST=$(oc get secret -n $ns $(oc get redis example-redis -n $ns -o jsonpath='{.status.secretRef.name}') -o jsonpath='{.data.uri}' | base64 --decode) -p PORT=6379 -p ENTRIES=<<no of entries>> -p USERS=<<no of concurrent users>> | oc create -f -
```

? or something like that, then we don't need to create and delete it

File: hack/redis/README.md (hunk 5 above)
Comment (ciaranRoche) [reply hack/redis/README.md]: Added in a comment in the doc.

File: hack/redis/redis-load/pkg/string/randomString.go [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+)
+
+const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+
+var seededRand = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+func stringWithCharset(length int, charset string) string {
+	b := make([]byte, length)
+	for i := range b {
+		b[i] = charset[seededRand.Intn(len(charset))]
```
Comment (aidenkeating) [inline hack/redis/redis-load/pkg/string/randomString.go:15]: Seeing

```
panic: runtime error: index out of range

goroutine 7 [running]:
math/rand.(*rngSource).Uint64(...)
	/usr/local/go/src/math/rand/rng.go:246
math/rand.(*rngSource).Int63(0xc420091500, 0x2517525b9a1f4709)
	/usr/local/go/src/math/rand/rng.go:231 +0x8a
math/rand.(*Rand).Int63(0xc4200801e0, 0x2517525b9a1f4709)
	/usr/local/go/src/math/rand/rand.go:82 +0x33
math/rand.(*Rand).Int31(0xc4200801e0, 0x2517525b)
	/usr/local/go/src/math/rand/rand.go:96 +0x2b
math/rand.(*Rand).Int31n(0xc4200801e0, 0x3e, 0xc400000015)
	/usr/local/go/src/math/rand/rand.go:131 +0x4f
math/rand.(*Rand).Intn(0xc4200801e0, 0x3e, 0x15)
	/usr/local/go/src/math/rand/rand.go:169 +0x45
github.com/ciaranRoche/redis-load/pkg/random.stringWithCharset(0x2a, 0x619245, 0x3e, 0x0, 0x0)
	/go/src/github.com/ciaranRoche/redis-load/pkg/random/randomString.go:15 +0xab
github.com/ciaranRoche/redis-load/pkg/random.RandomString(0x2a, 0x60e2dc, 0x5)
	/go/src/github.com/ciaranRoche/redis-load/pkg/random/randomString.go:21 +0x40
main.main.func1()
	/go/src/github.com/ciaranRoche/redis-load/main.go:44 +0x42f
github.com/ciaranRoche/redis-load/pkg/concurrency.ConcurrentFuncs.func1(0xc420016430, 0xc420010710)
	/go/src/github.com/ciaranRoche/redis-load/pkg/concurrency/concurrent.go:19 +0x4d
created by github.com/ciaranRoche/redis-load/pkg/concurrency.ConcurrentFuncs
	/go/src/github.com/ciaranRoche/redis-load/pkg/concurrency/concurrent.go:17 +0x20f
```

File: hack/redis/redis-load/main.go [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+		}
+		logger.Infof("connection made to %s", connectionString)
+		defer conn.Close()
+
+		count := 0
+		for count < options.entries {
+			entry := fmt.Sprintf("stuff:%s", croString.RandomString(42))
+
+			logger.Infof("adding stuff entry %s", entry)
+			_, err = conn.Do("HMSET", entry, "stuff", "all the stuff", "n thangs", "mo stuff", "no stuff", 42)
```
Comment (aidenkeating) [inline hack/redis/redis-load/main.go:47]: 👍

File: hack/redis/redis-load/pkg/string/randomString.go (hunk 6 above)
Comment (ciaranRoche) [reply hack/redis/redis-load/pkg/string/randomString.go]: I think i have fixed this, after testing with 1 million entries over 10 users, I am hitting 
```time="2020-08-06T10:49:41Z" level=fatal msg="OOM command not allowed when used memory > 'maxmemory'." action=makeNoise
```
essentially I reduced the charset to use a reduced number of bytes.

**Comment by aidenkeating** [comment]
:eyes:

**Pull request merged**